h2integrate/postprocess/mapping.py

```diff
@@ -0,0 +1,852 @@
+
+    # Load data
+    case_results_fpath = find_file(case_results_fpath)
+
+    # If case_results_fpath is a .csv file read in with pandas
+    if (".csv") in case_results_fpath.suffix:
+        results_df = pd.read_csv(case_results_fpath)
+    # Else if case_results_fpath is a path to the SQL recorder(s) (H2IntegrateModel.recorder_path)
+    # defined in the driver_config.yaml, read in with convert_sql_to_csv_summary()
+    elif (".sql") in case_results_fpath.suffix:
+        results_df = convert_sql_to_csv_summary(case_results_fpath, save_sql_file_to_csv)
+    else:
+        raise TypeError(
+            f"""The provided filepath {case_results_fpath} is of the wrong type, must be a .csv or
+            the .sql file defined in the driver_config.yaml (H2IntegrateModel.recorder_path)"""
+        )
+
+    # Auto detect latitude and longitude column names if not provided as argument
+    if latitude_var_name is None:
+        latitude_var_name, _ = auto_detect_lat_long_columns(results_df)
+
+    if longitude_var_name is None:
+        _, longitude_var_name = auto_detect_lat_long_columns(results_df)
+
+    # Create GeoDataFrame with results_df
+    results_gdf = gpd.GeoDataFrame(
+        results_df,
+        geometry=gpd.points_from_xy(
+            results_df[longitude_var_name],
+            results_df[latitude_var_name],
+            crs=map_preferences.lat_long_crs,
+        ),
+        crs=map_preferences.lat_long_crs,
+    )
+
+    # Convert coordinates to typical Web Mercator project CRS (EPSG:3857) for plotting
+    results_gdf = results_gdf.to_crs(map_preferences.web_map_crs)
+
+    # Create list variable to hold all gdfs for calculating basemap bounds
+    gdfs_for_bounds = [results_gdf]
+
+    # Check if fig, ax, and base_layer_gdf are all None
+    if all(v is None for v in [fig, ax, base_layer_gdf]):
+        # Then create new fig and ax objects (creating new map)
+        fig, ax = plt.subplots(
+            1,
+            figsize=map_preferences.figsize,
+            constrained_layout=map_preferences.constrained_layout,
+        )
+
+    # Else if fig, ax, and base_layer_gdf are all not None
+    elif all(v is not None for v in [fig, ax, base_layer_gdf]):
+        # Then set figure and axes as current fig, ax objects (adding layer to existing map)
+        plt.figure(fig.number)
+        plt.sca(ax)
+        # Validate base_layer_gdf(s) is in the same CRS as the results_gdf
+        base_layer_gdf = validate_gdfs_are_same_crs(base_layer_gdf, results_gdf)
+        # Extend gdf_for_bounds with base_layer_gdf(s) provided
+        gdfs_for_bounds.extend(base_layer_gdf)
+
+    # Else fig, ax, and base_layer_gdf not provided as a full set
+    else:
+        raise ValueError(
+            """The fig, ax, and base_layer_gdf arguments must be provided together to add a layer to
+            an existing plot or all must be omitted/None to create a new plot"""
+        )
```

Comment:
Instead of running the check loop multiple times, I would recommend the following control flow. I've also removed a few comments that are only stating what some of the straightforward pieces of code are doing without giving any additional context as to the why.

```suggestion
    # Ensure all of fig, ax, and base_layer_gdf are all None or have values provided, not mixed
    _check = [x is None for x in (fig, ax, base_layer_gdf)]
    if all(_check):
        fig, ax = plt.subplots(
            1,
            figsize=map_preferences.figsize,
            constrained_layout=map_preferences.constrained_layout,
        )
    elif any(_check):
        msg = (
            "The fig, ax, and base_layer_gdf arguments must be all provided together to add a layer to"
            "an existing plot or all must be omitted/None to create a new plot"
        )
        raise ValueError(msg)
    else:
        plt.figure(fig.number)
        plt.sca(ax)
        base_layer_gdf = validate_gdfs_are_same_crs(base_layer_gdf, results_gdf)
        gdfs_for_bounds.extend(base_layer_gdf)
```